src/Entity/ItemId.php

```diff
@@ -48,7 +50,11 @@ private function assertValidIdFormat( $idSerialization ) {
 	 * @return int
 	 */
 	public function getNumericId() {
-		return (int)substr( $this->serialization, 1 );
+		if ( $this->isForeign() ) {
+			throw new RuntimeException( 'getNumericId must not be called on foreign ItemIds' );
+		}
+
+		return (int)substr( $this->getSerialization(), 1 );
```

Comment:
Do we know this is normalized and does not start with a colon? I do miss this line of code somehow, or don't see it.